entgql/internal/todo/todo_test.go

```diff
@@ -640,6 +640,118 @@ func (s *todoTestSuite) TestPaginationFiltering() {
 	})
 }
 
+func (s *todoTestSuite) TestEdgesFiltering() {
+	ctx := context.Background()
+	tr := s.ent.Todo.Create().
+		SetText("t1").
+		SetStatus(todo.StatusInProgress).
+		SaveX(ctx)
+
+	t1 := s.ent.Todo.Create().
+		SetText("t2 0").
+		SetStatus(todo.StatusInProgress).
+		SetParent(tr).
+		SaveX(ctx)
+	s.ent.Todo.Create().
+		SetText("t2 1").
+		SetStatus(todo.StatusCompleted).
+		SetParent(tr).
+		SaveX(ctx)
+
+	t2 := s.ent.Todo.Create().
+		SetText("t3 0").
+		SetStatus(todo.StatusInProgress).
+		SetParent(t1).
+		SaveX(ctx)
+	t3 := s.ent.Todo.Create().
+		SetText("t3 1").
+		SetStatus(todo.StatusInProgress).
+		SetParent(t1).
+		SaveX(ctx)
+
+	query := `query todos($id: ID!, $lv2Status: TodoStatus!) {
+		todos(where:{id: $id}) {
+			edges {
+				node {
+					childrenLevel0: children(where: {statusNEQ: COMPLETED}) {
+						totalCount
+						edges {
+							node {
+								id
+								childrenLevel1: children(where: {statusNEQ: $lv2Status}) {
```

Comment:
OK, I see now. It's called `childrenLevel1` and that's why it's not loaded. 

I still want to give it a look before merging it.